python/ray/tune/schedulers/trial_scheduler.py

```diff
@@ -11,6 +11,7 @@ class TrialScheduler(object):
     CONTINUE = "CONTINUE"  #: Status for continuing trial execution
     PAUSE = "PAUSE"  #: Status for pausing trial execution
     STOP = "STOP"  #: Status for stopping trial execution
+    NOOP = "NOOP"  #: Status for no-op (letting scheduler manage)
```

Comment:
I don't get this, the scheduler is returning NOOP, so who's making the decision then?

Reply:
The scheduler has access to the executor - for example if there are two trials, one of which you want to kill and the other you specifically want to subsume the resources of the first trial, returning "Trial.STOP" wouldn't work. Instead, all of this should be handled in the `scheduler.on_trial_result`, and the trial_runner doesn't have an explicit decision to make.

Reply:
The decision is CONTINUE isn't it?

The exact situation is handled already in PBT which resets trial params on the fly: https://github.com/ray-project/ray/blob/master/python/ray/tune/schedulers/pbt.py#L235

We could allow reset() to take new resources too.

Reply:
No. To clarify - the procedure is triggered by the return result of Trial 1; calling `CONTINUE` on Trial 1 (which you've killed) would result in error...

Reply:
Why don't you just return STOP instead of killing it?